src/arm.rs

```diff
@@ -1,3 +1,35 @@
+use core::intrinsics;
+
```

Comment:
A comment explaining why an asm implementation was necessary (non-standard calling convention) would be nice here.